### ISSUE TITLE
Improve Core upgrader

### ISSUE THREAD

**Issue by craigh** [zikula/core#2833]
| Q | A |
| --- | --- |
| Zikula Version | 1.4.2 |
| PHP Version | ~ |

The core upgrader is still not flexible enough. It appears to always upgrade from 1.3.x . This must be improved before 1.4.3

**Comment by rallek** [comment]
The complete process should be reviewed. It would be great if we have to delete a complete folder but all files which have to be rescued should be in another folder. That would make things more simple. But the complete structure of our installations must be restructured than.

What I did manually: I have my zikula installations allways inside the folder Zikula_Core-1.4.0 (or the right core number) what is comming out of the tar file. So in case of upgrade from 1.4.0 to 1.4.2 I do unpack the new core to Zikula_Core-1.4.2. Than I copied my configuration files (custom_parameters.yml, config.php and personal_config.php), /modules, /userdata and my own theme. After finishing I also have to change my configuration where the server should look for my site.

Might be this will give an input for an idea of the upgrade process.
